client/my-sites/media-library/list-item.tsx

```diff
@@ -88,7 +96,11 @@ export default class extends React.Component {
 	};
 
 	renderSpinner = () => {
-		if ( ! this.props.media || ! this.props.media.transient ) {
+		if ( ! this.props.media ) {
+			return;
+		}
+
+		if ( ! ( this.props.media as MediaObject ).transient ) {
```

Comment:
Just curious - supposedly this is the same as (?):

```
const { media } = this.props;

if ( ! ( media && ( media as MediaObject ).transient ) ) {
  return;
}
```

Looks more succinct and still readable if so, but I may be wrong.

Reply:
NIT: I assume you tried to stay reasonably close to what was previously implemented, but if we change this I'd prefer this to be inlined in the main `render` and remove this method entirely:

```tsx
{ media && ( media as MediaObject ).transient && (
  <div className="media-library__list-item-spinner">
    <Spinner />
  </div>
) }
```

Reply:
Thank you both! That is indeed simpler and not something I'd looked at critically, as I mostly just went through the motions converting the file. Went with @sirreal's option as the simplest.